onedal/linear_model/linear_model.py

```diff
@@ -293,15 +293,15 @@ def fit(self, X, y, queue=None):
         """
         module = self._get_backend("linear_model", "regression")
 
-        X = _check_array(
-            X,
-            dtype=[np.float64, np.float32],
-            force_all_finite=False,
-            ensure_2d=False,
-            copy=self.copy_X,
-        )
+        if not isinstance(X, np.ndarray):
```

Comment:
@samir-nasibli  Is there a ticket for Ridge Array_api support? I know you had mentioned you wanted this to be more in the direction of array_api, but I think maybe it should be split into 2 PRs - out of preview and array_api support.  If thats okay, could you make sure to send @DDJHB and me the ticket, or we can have @DDJHB make it.  You can see this code is rather numpy-oriented.

Reply:
@samir-nasibli ping

Reply:
Currently only numpy.ndarray array inputs are expected. No need for branching. I think now this branching is for avoiding sparse inputs.